src/main/kotlin/io/zerobase/smarttracing/GraphDao.kt

```diff
@@ -53,4 +57,183 @@ class GraphDao(private val driver: Driver) {
+                        """.trimIndent()
+                ).single()["id"].asString()
+                return@writeTransaction result?.let { OrganizationId(it) }
+            }
+        }
+    }
+
+    /**
+     * Sets the multi-site flag in an organization.
+     *
+     * @param id organization uuid.
+     * @param state the value for the multi-site flag
+     */
+    fun setMultiSite(id: String, state: Boolean) {
+        driver.session().use {
+            it.writeTransaction { txn ->
+                txn.run(
+                        """
+                        MATCH (o:Organization { id: '${id}' })
+                        SET o.multisite = 'true'
+                        """
+                )
+            }
+        }
+    }
+
+    /**
+     * Creates site.
+     *
+     * @param id organization id.
+     * @param name name of the site.
+     * @param category category of the site.
+     * @param subcategory subcategory of the site.
+     * @param lat latitude of the site.
+     * @param long longitude of the site.
+     * @param testing whether the site can preform testing.
+     * @param phone contact phone of site manager
+     * @param email contact email of site manager
+     * @param contactName contact name of site manager
+     */
+    fun createSite(id: String, name: String, category: String,
+                   subcategory: String, lat: Float, long: Float,
+                   testing: Boolean, phone: String?, email: String?,
+                   contactName: String?): SiteId? {
+        val query = if (phone == null) {
+            """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: o.phone,
+                        email: o.email,
+                        contactName: o.contactName,
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        } else {
+             """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: '${phone!!}',
+                        email: '${email!!}',
+                        contactName: '${contactName!!}',
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        }
```

Comment:
there's no need to have to variations of the query. just use the elvis operator the provide a default value if you don't want the field to be null in the database: `phone: '${phone ?: "none"}'`

Reply:
the reason for that is because if it is null we use the default one on the organization, so it is not none, i did the !! in order to make sure it was a string when it should be one if phone is not null, i was thinking of using '${phone ?: "o.phone"}' to minimize it, but the issue there is the ' ' around o.phone makes it a literal instead of a copy of o.phone

Reply:
just set the field to `null` in the database if you don't want a default. `null` properties are removed from a node.

Reply:
alright will do that